src/main/java/org/broadinstitute/consent/http/service/dao/DatasetServiceDAO.java

```diff
@@ -67,15 +67,17 @@ public void deleteDataset(Dataset dataset, Integer userId) throws Exception {
   public void deleteStudy(Study study, User user) throws Exception {
     jdbi.useHandle(handle -> {
       handle.getConnection().setAutoCommit(false);
-      study.getDatasets().forEach(d -> {
-        try {
-          deleteDataset(d, user.getUserId());
-        } catch (Exception e) {
-          handle.rollback();
-          logException(e);
-          throw new DatasetDeletionException(e);
-        }
-      });
+      if (study.getDatasets() != null) {
```

Comment:
Could you instead fix this by changing `Study` to initialize `datasets` so it's not `null`? That would avoid the need to check for `null` everywhere this getter is used.
```
  private Set<Dataset> datasets = Set.of();
```